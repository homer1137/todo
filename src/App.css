.App {
  background-color: rgb(245,245,245);
  height: 100vh;
  width: 100vw;
}

.todo-title{
  background-color: rgb(255,255,255);
  
  height: 60px;
  border: 2px solid rgb(235,235,235);
  display: flex;
  align-items: center;
  padding: 20px;
  box-shadow: 2px 2px 2px rgb(235,235,235);
}


section {
  max-width: 700px;
  margin: 0 auto;
  padding: 20px;
}

h1{
  color: rgb(233,217,216);
  font-size: 7rem;
  text-align: center;
  font-weight: 100;
}
h2{
  color: rgb(230,230,230);
  font-size: 2rem;
  font-weight: 300;
  font-style: italic
}


