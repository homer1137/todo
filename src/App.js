import React from "react";
import { useState } from "react";
import "./App.css";
import Todo from "./components/Todo/Todo";
import FormCreateTodo from "./components/FormCreateTodo/FormCreateTodo";
import ManageTodo from "./components/ManageTodo/ManageTodo";

function App() {
  const [todo, setTodo] = useState([
    { id: 1, text: "hello", done: false },
    { id: 2, text: "hi again", done: false },
    { id: 3, text: "to do task", done: false },
  ]);

  const [active, setActive] = useState(false);
  const [completed, setCompleted] = useState(false);

  const sortedPosts = () => {
    if (active) return [...todo].filter((item) => item.done === false);
    if (completed) return [...todo].filter((item) => item.done === true);
    else return todo;
  };

  function doneToDo(currentDodo) {
    setTodo(
      todo.map((td) => {
        if (td.id === currentDodo.id) {
          let done = !td.done;
          return { ...td, done: done };
        } else return { ...td };
      })
    );
  }

  function deleteDone() {
    setTodo(todo.filter((td) => td.done !== true));
  }

  let result = todo.filter((item) => item.done !== true).length;

  return (
    <div className="App">
      <section>
        <h1>todos</h1>
        <div>
          <h2 className="todo-title">What is need to be done</h2>
          {todo.length>0?sortedPosts().map((item, index) => (
            <Todo
              todo={item}
              todoIndex={index+1}
              doneToDo={doneToDo}
              key={item.id}
            />
          )):<h2>There are no todo. Create a todo</h2>}
        </div>
      <ManageTodo
        result={result}
        setActive={setActive}
        setCompleted={setCompleted}
        active={active}
        completed={completed}
        deleteDone={deleteDone}
      />
      <FormCreateTodo setTodo={setTodo} />
      </section>
    </div>
  );
}

export default App;
