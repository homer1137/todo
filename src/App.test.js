import { toBeInTheDocument } from "@testing-library/jest-dom/dist/matchers";
import { render, screen, fireEvent } from "@testing-library/react";
import App from './App'

 //наличие элементов на странице

describe('Наличие элементов на странице', ()=>{
    const setup = () => render(<App/>);
    
    test('Заголовок Todos', ()=>{
        setup();
        const h1element = screen.getByText(/todos/i)
        expect(h1element).toBeInTheDocument();
    })
    test('Заголовок h2', ()=>{  
        setup();
        const h2element = screen.getByText(/What is need to be done/i)
        expect(h2element).toBeInTheDocument();
     })
     test('Заголовок inputCreateTodo', ()=>{  
        setup();
        const inputCreateTodo = screen.getByPlaceholderText(/Todo description/i)
        expect(inputCreateTodo).toBeInTheDocument();
     })
     test('Заголовок buttonCreateTodo', ()=>{  
        setup();
        const buttonCreateTodo = screen.getByText(/Create new todo/i)
        expect(buttonCreateTodo).toBeInTheDocument();
     })
     test('Заголовок buttonAll', ()=>{  
        setup();
        const buttonAll = screen.getByText(/All/i)
        expect(buttonAll).toBeInTheDocument();
     })
     test('Заголовок buttonActive', ()=>{  
        setup();
        const buttonActive = screen.getByText(/Active/i)
        expect(buttonActive).toBeInTheDocument();
     })
     test('Заголовок buttonCompleted', ()=>{  
        setup();
        const buttonCompleted = screen.getByText('Completed')
        expect(buttonCompleted).toBeInTheDocument();
     })
     test('Заголовок buttonClearCompleted', ()=>{ 
        setup(); 
        const buttonClearCompleted = screen.getByText(/Clear Completed/i)
        expect(buttonClearCompleted).toBeInTheDocument();
        expect(buttonClearCompleted).toMatchSnapshot()
     })
     test('Заголовок items left', ()=>{  
        setup();
        const itemsLeft = screen.getByText(/items left/i)
        expect(itemsLeft).toBeInTheDocument();
        expect(itemsLeft).toMatchSnapshot()
     })
    
})

//Создание дела

describe('create todo', ()=>{
    
    test('create todo', ()=>{
        render(<App/>);
        const inputCreateTodo = screen.getByPlaceholderText(/Todo description/i);
        expect(inputCreateTodo).toContainHTML('')
        fireEvent.input(inputCreateTodo, {target:{value: 'say hello'}})
        expect(inputCreateTodo).toContainHTML('say hello')
        const buttonCreateTodo = screen.getByText(/Create new todo/i)
        fireEvent.click(buttonCreateTodo);
        const newTodo = screen.getByText(/say hello/i)
        expect(newTodo).toBeInTheDocument();
    })
})


