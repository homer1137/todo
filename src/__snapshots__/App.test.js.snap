// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Наличие элементов на странице Заголовок buttonClearCompleted 1`] = `
<button
  class="myBtn"
  style="color: rgb(169, 169, 169); font-weight: 300; font-size: 1.3rem;"
>
  Clear completed
</button>
`;

exports[`Наличие элементов на странице Заголовок items left 1`] = `
<div>
  3
   items left
</div>
`;
